src/ConnectedMode.UnitTests/QualityProfiles/QualityProfileDownloaderTests.cs

```diff
@@ -53,6 +52,31 @@ public void MefCtor_CheckIsSingleton()
         {
             MefTestHelpers.CheckIsSingletonMefComponent<QualityProfileDownloader>();
         }
+
+        [TestMethod]
+        public async Task UpdateAsync_NothingToUpdate_ReturnsFalse()
+        {
+            var boundSonarQubeProject = CreateBoundProject();
+            SetupLanguagesToUpdate(out var outOfDateQualityProfileFinderMock,
+                boundSonarQubeProject,
+                Array.Empty<Language>());
+            var logger = new TestLogger();
+            var solutionBindingOperationMock = new Mock<ISolutionBindingOperation>();
+
+            var testSubject = CreateTestSubject(outOfDateQualityProfileFinderMock.Object,
+                logger: logger,
+                solutionBindingOperation: solutionBindingOperationMock.Object,
+                languagesToBind: Language.KnownLanguages.ToArray());
+
+            var result = await testSubject.UpdateAsync(boundSonarQubeProject, null, CancellationToken.None);
+
+            result.Should().BeFalse();
+            solutionBindingOperationMock.Verify(
```

Comment:
The check here could be simpler and easier to read e.g. `solutionBindingOperationMock.Invocations.Should().BeEmpty()`.

But I wouldn't bother changing it since I am going to delete the class completely in one of my PRs.